src/shared/exec-util.c

```diff
@@ -156,6 +156,17 @@ static int do_execute(
                         log_debug("About to execute %s%s%s", t, argv ? " " : "", argv ? strnull(args) : "");
                 }
 
+                if (FLAGS_SET(flags, EXEC_DIR_WARN_WORLD_WRITABLE)) {
+                        struct stat st;
+
+                        r = stat(t, &st);
+
+                        if (r < 0)
+                                log_warning_errno(errno, "Failed to stat %s: %m", t);
+                        else if (S_ISREG(st.st_mode) && (st.st_mode & 0002))
+                                log_warning("%s is world writable. This is not a good idea.", t);
```

Comment:
Maybe expand a little to match a similar warning that already exists elsewhere: `"%s is marked world-writable, which is a security risk as it is executed with privileges. Please remove world writability permission bits. Proceeding anyway.`